client/log_power.py

```diff
@@ -30,6 +30,35 @@ def read_measurement():
     sp.check_returncode()
     return sp.stdout
 
+def read_rpm(threshold_value):
+
+    set_th_rising = subprocess.run([f"{DIRECTORY}/micromini-tool", "set-ain-rising-threshold", threshold_value], capture_output=False)
+    set_th_rising.check_returncode()
+    set_th_falling = subprocess.run([f"{DIRECTORY}/micromini-tool", "set-ain-falling-threshold", threshold_value], capture_output=False)
+    set_th_falling.check_returncode()
+
+    sp_rising = subprocess.run([f"{DIRECTORY}/micromini-tool", "get-ain-nrising"], capture_output=True)
+    sp_rising.check_returncode()
+    sp_falling = subprocess.run([f"{DIRECTORY}/micromini-tool", "get-ain-nfalling"], capture_output=True)
+    sp_falling.check_returncode()
+    sp_samples = subprocess.run([f"{DIRECTORY}/micromini-tool", "get-ain-num-samples"], capture_output=True)
+    sp_samples.check_returncode()
+    sp_sampling_rate = subprocess.run([f"{DIRECTORY}/micromini-tool", "get-ain-rate"], capture_output=True)
+    sp_sampling_rate.check_returncode()
+
+    rising_crossings = float(sp_rising.stdout.decode("utf-8").strip().split(":")[1].strip())
+    falling_crossings = float(sp_falling.stdout.decode("utf-8").strip().split(":")[1].strip())
+    num_samples = float(sp_samples.stdout.decode("utf-8").strip())
+    sampling_rate = float(sp_sampling_rate.stdout.decode("utf-8").strip().split(":")[1].strip())
+
+    seconds_per_minute = 60
+    crossings_per_rotation = 5
```

Comment:
where is this number coming from (5 blades ?)

Reply:
It's the factor to convert electrical cycles to mechanical revolutions, it's inherit to the generator we use on the turbines.